WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -692,6 +692,10 @@ private extension ProductFormViewModel {
 
 private extension ProductFormViewModel {
     func updateBlazeEligibility() {
+        guard formType == .edit else {
+            isEligibleForBlaze = false
+            return
+        }
```

Comment:
> No unit test was added because the check was added in a class conforming to a protocol, so it was not very testable unless we refactor how the product form works. Please feel free to share any suggestions for testing this.

I think we can add a test case similar to [this](https://github.com/woocommerce/woocommerce-ios/blob/e46ab13c7c8ea7388416a9f5ba20c62b0d30b1c9/WooCommerce/WooCommerceTests/ViewRelated/Products/Edit%20Product/ProductFormViewModelTests.swift#L183):

```swift
    func test_canPromoteWithBlaze_is_false_for_product_creation_when_product_is_eligible_for_blaze() {
        // Given
        let product = Product.fake()
        let blazeEligibilityChecker = MockBlazeEligibilityChecker(isProductEligible: true)
        let nonEditableFormTypes: [ProductFormType] = [.add, .readonly]

        // When
        nonEditableFormTypes.forEach { nonEditableFormType in
            // When
            let viewModel = createViewModel(product: product, formType: nonEditableFormType, blazeEligibilityChecker: blazeEligibilityChecker)

            waitUntil {
                // Then
                viewModel.canPromoteWithBlaze() == false
            }
        }
    }
```

Reply:
Good idea, added in 25674e7030823943ba06e4312f998d489cda9e3b.